database/migrations/create_chat_tables.php

```diff
@@ -28,145 +27,66 @@ public function __construct()
-                $table->integer('user_id')->unsigned();
-                $table->string('type')->default('text');
-                $table->timestamps();
-
-                $table->foreign('user_id')
-                    ->references($this->userModelPrimaryKey)
-                    ->on($this->userModelTable)
-                    ->onDelete('cascade');
-
-                $table->foreign('conversation_id')
-                    ->references('id')
-                    ->on('mc_conversations')
-                    ->onDelete('cascade');
-            });
-
-            Schema::create('mc_conversation_user', function (Blueprint $table) {
-                $table->integer('user_id')->unsigned();
-                $table->integer('conversation_id')->unsigned();
-                $table->primary(['user_id', 'conversation_id']);
-                $table->timestamps();
-
-                $table->foreign('conversation_id')
-                    ->references('id')->on('mc_conversations')
-                    ->onDelete('cascade');
-
-                $table->foreign('user_id')
-                    ->references($this->userModelPrimaryKey)
-                    ->on($this->userModelTable)
-                    ->onDelete('cascade');
-            });
-
-            Schema::create('mc_message_notification', function (Blueprint $table) {
-                $table->increments('id');
-                $table->integer('message_id')->unsigned();
-                $table->integer('conversation_id')->unsigned();
-                $table->integer('user_id')->unsigned();
-                $table->boolean('is_seen')->default(false);
-                $table->boolean('is_sender')->default(false);
-                $table->boolean('flagged')->default(false);
-                $table->timestamps();
-                $table->softDeletes();
-
-                $table->index(['user_id', 'message_id']);
-
-                $table->foreign('message_id')
-                    ->references('id')->on('mc_messages')
-                    ->onDelete('cascade');
-
-                $table->foreign('conversation_id')
-                    ->references('id')->on('mc_conversations')
-                    ->onDelete('cascade');
-
-                $table->foreign('user_id')
-                    ->references($this->userModelPrimaryKey)
-                    ->on($this->userModelTable)
-                    ->onDelete('cascade');
-            });
-        }
+        Schema::create('mc_conversations', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->boolean('private')->default(true);
+            $table->text('data')->nullable();
+            $table->timestamps();
+        });
+
+        Schema::create('mc_messages', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->text('body');
+            $table->bigInteger('conversation_id')->unsigned();
+            $table->bigInteger('user_id')->unsigned();
```

Comment:
This needs to be changed too, right?

Reply:
Seems we have a few more updates to make. Updating now

Reply:
Thanks a lot! :) Your are a busy man 😄

Reply:
If you need, send me an email to `tinashemusonza` gmail and i will add you to a slack channel so you can help me with completing the updates. Thanks